src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -132,10 +166,45 @@ public void pcre2_versioncheck(){
 //        compile_options = new LibJavaPcre.OptionsStruct();
 //    }
 
-    public void pcre2_compile_java(String pat, int size){
+    public void pcre2_gcontext_create(){
+        gcontext = LibJavaPcre.INSTANCE.pcre2_gcontext_create();
+    }
+    public void pcre2_ccontext_create(){
+        if (gcontext == null){
+            LOGGER.error("General context is not initialized properly!");
+        } else {
+            ccontext = LibJavaPcre.INSTANCE.pcre2_ccontext_create(gcontext);
+        }
+    }
+    public void pcre2_ccontext_set_extra_options(){
+        LibJavaPcre.INSTANCE.pcre2_ccontext_set_extra_options(ccontext, extra_options);
+    }
+    public void pcre2_mcontext_create(){
+        if (gcontext == null){
+            LOGGER.error("General context is not initialized properly!");
```

Comment:
IllegalStateException might be better here

Reply:
Added the exception